src/basic/macro.h

```diff
@@ -112,8 +112,23 @@
 
 /* Rounds up */
 
-#define ALIGN4(l) (((l) + 3) & ~3)
-#define ALIGN8(l) (((l) + 7) & ~7)
+#define _ALIGN(l, ali)                                          \
+        ({                                                      \
+                typeof(l) _l = -1;                              \
+                                                                \
+                assert_cc(ali == 4 || ali == 8);                \
+                assert(_l > 0); /* check unsigned */            \
```

Comment:
`assert_cc(IS_UNSIGNED_INTEGER_TYPE(typeof(l));`

Or is this supposed to allow signed types?

Reply:
Also, this seems to do the exact same thing as ALIGN_TO. Maybe just call that?